lib/index.js

```
@@ -0,0 +1,434 @@
+                id: executionId,
+                index: executions.length,
+                branch: 0,
+                type: 'request',
+                model: {
+                    $self: ['requests', data.cursor.httpRequestId]
+                },
+                cursor: data.cursor,
+                context: {
+                    ref: data.cursor.ref,
+                    httpRequestId: data.cursor.httpRequestId,
+                    scriptId: data.cursor.scriptId,
+                    execution: data.cursor.execution
+                },
+                run: {
+                    $self: ['run', runId]
+                },
+                error: err && { $self: ['errors', err.id] },
+                timings: {
+                    started: requestStartTime,
+                    completed: Date.now()
+                }
+            });
+
+            requests[data.cursor.httpRequestId] = {
+                id: data.cursor.httpRequestId,
+                execution: {
+                    $self: ['executions', executionId]
+                },
+                response: {
+                    $self: ['responses', data.respons && data.response.id]
+                },
+                url: data.request.url.toString(),
+                method: data.request.method
+            };
+
+            // If there was no response then do not add it
+            if (!data.response) {
+                return;
+            }
+
+            responses[data.response.id] = {
+                request: {
+                    $self: ['requests', data.cursor.httpRequestId]
+                },
+                id: data.response.id,
+                status: data.response.status,
+                code: data.response.code,
+                responseTime: data.response.responseTime,
+                responseSize: data.response.responseSize
+            };
+        });
+
+        newman.on('assertion', function (err, data) {
+            var executionId = uuid(),
```

Need a comment explaining why errors are not being handled here